### DISCUSSION TITLE
Could `data-repo-id` and `data-category-id` be optional?

### DISCUSSION THREAD

**Discussion by razonyang** [giscus/giscus#853] [Q&A]
Hi, I'm wondering if it's possible to simplify the configuration that remove `data-repo-id` and `data-category-id`, and fetching those IDs in server side by names.

Thanks your awesome works.

**Comment by pawamoy** [comment]
I second that. I'm trying to integrate Giscus in a project template, all I know in advance is the account and repository names, like `pawamoy/testing`. If I wanted to fetch the IDs automatically, I'd need some kind of post-process hook that sends requests to GitHub and replaces placeholders in a HTML file. Since account and repository names are unique on GitHub (unless I'm missing something), couldn't we get rid of these ids which make automation much harder :thinking:?

Same for category ID: I created the same category (same title, same description, same icon) in two different repositories and they got two different IDs, so it's once again hard to automate the integration.

The need to somehow fetch these IDs also means my template doesn't work offline anymore.

By the way, I tried simply removing these fields, but I get this error when posting a comment:

```
TypeError: (destructured parameter).data is undefined
```

In https://giscus.app/_next/static/chunks/pages/widget-540154a6663551e0.js.
